serve/mlc_serve/model/model_common.py

```diff
@@ -75,13 +75,22 @@ def _is_safe_to_sample(prob_like):
     logits = torch.from_dlpack(logits)
     num_seq = len(sampling_params)
 
-    mask_random = torch.tensor(
+    mask_random_cpu = torch.tensor(
         [p.sampling_type == SamplingType.RANDOM for p in sampling_params],
-        dtype=torch.bool,
+        dtype=torch.bool
     )
-    mask_greedy = torch.logical_not(mask_random)
+    mask_greedy_cpu = torch.logical_not(mask_random_cpu)
+    if logits.device == torch.device("cpu"):
```

Comment:
Don't need this case.

Reply:
It costs nothing but we support case with run topology on CPU. As I know @elvin-n run singlebatch on CPU after small fixes